lib/diagnostic.rb

```diff
@@ -5,7 +5,13 @@ class Diagnostic
   # enumerable supplied.  Assume each entry implements a length
   # method that returns a Fixnum.
   def longest(enum)
-    # Your code goes here
+    longest = enum.first
+    enum.each do |entry|
+      if entry.length > longest.length
+        longest = entry
+      end
+    end
+    longest
```

Comment:
This will work, but Enumerable specifically provides us with tools to address situations like this: `max` and `max_by`.
e.g.
`enum.max_by { |entry| entry.length }` or `enum.max_by(&:length)`
`enum.max { |a, b| a.length <=> b.length }`